src/libsync/networkjobs.cpp

```diff
@@ -346,6 +346,8 @@ const QHash<QString, qint64> &PropfindJob::sizes() const
 AvatarJob::AvatarJob(AccountPtr account, const QString &userId, int size, QObject *parent)
     : AbstractNetworkJob(account, account->url(), QStringLiteral("remote.php/dav/avatars/%1/%2.png").arg(userId, QString::number(size)), parent)
 {
+    setStoreInCache(true);
```

Comment:
Requiring two calls looks like a good source for bugs?

Reply:
I don't think just a subset of all possible permutations of these flags will suit our needs. We need the flexibility, though. The future network jobs added to download the server theme and according resources need to load both explicitly from the cache only(!) as well as download resources from the network (while storing this fresh response in the cache if successful). We can implement this by creating a the job twice and changing the caching policies accordingly.

I documented that both functions are related to one another. Also, the one existing example of cached resources, user avatars, uses `PreferCache` with `true` for instance. The approach attempting downloading and using the cache in parallel mentioned above will combine `AlwaysCache` with `false` and `AlwaysNetwork` with `true`.